### DISCUSSION TITLE
active learning

### DISCUSSION THREAD

**Discussion by GHLau** [ACEsuit/mace#657] [Q&A] [answered]
Dear developer，

How to use the active learning? Can you have a little guide on how to use it?

Thank you so much!

**Comment by gabor1** [comment, marked as answer]
Dear user,

This is a rather general question, not specific to the mace code. The general idea is that you run some simulations using an existing model (eg mace-mp0) and you collect configurations that you want more accuracy on. Either because you test them and you see that errors are not good enough, or simply they occur in your simulations, or because the model ensemble disagreement is high (there are extensions in mace that can estimate the model uncertainty, or you use multiple models and look at their disagreement). Then you compute these new structures with your favourite qm method and you train a new model that includes this data.

These ideas are described in some review papers, e.g. in Deringer et al. Chemical Reviews 2021

**Reply by GHLau** [reply]
ok, thank you for your guidance.  Is the 'mace/cli/active_learning_md.py' an example?

**Reply by ilyes319** [reply]
Hey @GHLau, indeed it is one example using the forces disagreement of a committee of models. You can directly construct a committee in the MACE calculator using a list of paths. Then use the variance of their forces or any other metric to select your new configuration during MD.
